src/Services/PractitionerRoleService.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/**
+ * PractitionerRoleService
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Yash Bothra <yashrajbothra786gmail.com>
+ * @copyright Copyright (c) 2018 Matthew Vita <matthewvita48@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Services;
+
+use OpenEMR\Common\Uuid\UuidRegistry;
+use OpenEMR\Validators\ProcessingResult;
+use OpenEMR\Validators\PractitionerRoleValidator;
+
+class PractitionerRoleService extends BaseService
+{
+
+    private const PRACTITIONER_ROLE_TABLE = "facility_user_ids";
+    private $practitionerRoleValidator;
+    private $uuidRegistery;
+
+    /**
+     * Default constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct('facility_user_ids');
+        $this->uuidRegistery = new UuidRegistry(['table_name' => self::PRACTITIONER_ROLE_TABLE]);
+        $this->uuidRegistery->createMissingUuids();
```

Comment:
also, since this will need uuid population in users and facility, I'm guessing you'll need to ensure these are populated. maybe something like:
```
(new UuidRegistry(['table_name' => 'users']))->createMissingUuids();
(new UuidRegistry(['table_name' => 'facility']))->createMissingUuids();

```